lib/ansible/module_utils/network/routeros/routeros.py

```diff
@@ -0,0 +1,148 @@
+# This code is part of Ansible, but is an independent component.
+# This particular file snippet, and this file snippet only, is BSD licensed.
+# Modules you write using this snippet, which is embedded dynamically by Ansible
+# still belong to the author of the module, and may assign their own license
+# to the complete work.
+#
+# (c) 2016 Red Hat Inc.
+#
+# Redistribution and use in source and binary forms, with or without modification,
+# are permitted provided that the following conditions are met:
+#
+#    * Redistributions of source code must retain the above copyright
+#      notice, this list of conditions and the following disclaimer.
+#    * Redistributions in binary form must reproduce the above copyright notice,
+#      this list of conditions and the following disclaimer in the documentation
+#      and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+# USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+import json
+from ansible.module_utils._text import to_text
+from ansible.module_utils.basic import env_fallback, return_values
+from ansible.module_utils.network.common.utils import to_list, ComplexList
+from ansible.module_utils.connection import Connection
+
+_DEVICE_CONFIGS = {}
+
+routeros_provider_spec = {
```

Comment:
Provider spec is required to support connection=local type and is present in other network platform for legacy reason, in future local connection type for network module will be deprecated. 
As this platform is newly added supporting local connection type is not required imo and this spec can be removed.

Reply:
I am planning on implementing support for RouterOS API as well. I could implement it either with a provider dict, or by creating a special connection type. Which do you think would be better?

RouterOS has a weird non-standard API, so it seems that it’s not possible to use any of the existing connection types, except for local, of course.

Reply:
Which transport is type is used by RouterOS API? 
You can check [httapi](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/connection/httpapi.py) connection type and implementation plugins for [nxap/eapi](https://github.com/ansible/ansible/tree/devel/lib/ansible/plugins/httpapi).

Reply:
In that case, it seems connection=local is the way forward for now. You can ignore my comment on providers.